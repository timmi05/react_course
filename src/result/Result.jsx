import React, {Component} from 'react';
import {ResultPanel} from './panel/ResultPanel';
import {ResultBody} from './body/ResultBody';
import {fetchMovies, openMovie} from '../store/actions/movies';
import {connect} from 'react-redux';
import {selectMovies} from '../store/selectors/selectMovies';
import {selectIsProductSelected} from '../store/selectors/selectIsProductSelected'
import {selectSelectedProduct} from '../store/selectors/selectSelectedProduct'
import {findMovie} from '../services/filterMovies'
import {selectGenre} from '../store/selectors/selectGenre'

class Result extends Component {
    componentDidMount() {
        this.props.fetchMovies();
    };

    openMovie = id => {
        this.props.onOpenMovie(findMovie(this.props.movies, id));
    };

    render() {
        const movies = this.props.movies;
        return (
            <div className='result'>
                <ResultPanel
                    isProductSelected={this.props.isProductSelected}
                    genre={this.props.genre}
                    length={movies.length}
                />
                <ResultBody
                    movies={movies}
                    openMovie={this.openMovie}
                />
            </div>
        )
    };
}


const mapStateToProps = state => {
    return ({
        movies: selectMovies(state),
        isProductSelected: selectIsProductSelected(state),
        genre: selectGenre(selectSelectedProduct(state))
    });
};


const mapDispatchToProps = dispatch => ({
    onOpenMovie: movie => dispatch(openMovie(movie)),
    fetchMovies: () => dispatch(fetchMovies())
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Result);
