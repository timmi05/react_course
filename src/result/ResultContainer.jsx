import React from 'react';
import ResultPanel from './panel/ResultPanel';
import ResultBody from './body/ResultBody';

const ResultContainer = (props) => {
    return(
     <div className="result-container">
         <ResultPanel
         {...props}
          />
         <ResultBody/>
    </div>
    )
}
export default ResultContainer