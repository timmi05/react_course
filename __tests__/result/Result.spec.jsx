import React from 'react';
import { shallow } from 'enzyme';
import { Result } from '../../src/result/Result';

describe('Result rendering', () => {
    it('renders correctly', () => {
        const component = shallow(<Result/>);
        expect(component).toMatchSnapshot();
    });
});
