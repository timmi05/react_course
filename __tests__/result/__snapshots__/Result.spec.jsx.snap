// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result rendering renders correctly 1`] = `
<div
  className="result"
>
  <ResultPanel />
  <ResultBody />
</div>
`;
